src/DAL/connectionManager.ts

```diff
@@ -27,7 +26,7 @@ export class ConnectionManager {
     } catch (err) {
       const errString = JSON.stringify(err, Object.getOwnPropertyNames(err));
       this.logger.error(`failed to connect to database: ${errString}`);
-      throw new DBConnectionError();
```

Comment:
this should be removed but based on the new error package.
custom error types can be used to filter exceptions